### PR TITLE
[BLOCKED] Server properties

### PR DESCRIPTION
**Pull request by muhrin**
Exposes information about the RMQ server to communicator clients.  Could be useful for AiiDA as this allows the version number to be determined.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Looks good, I'll wait for you to work out why all the tests are failing 😅

File: kiwipy/rmq/communicator.py [hunk 1]
```diff
@@ -345,6 +345,13 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
     def __str__(self):
         return f'RMQCommunicator({self._connection})'
 
+    @property
+    def server_properties(self) -> Dict:
```
Comment (chrisjsewell) [inline kiwipy/rmq/communicator.py:349]: maybe add a docstring

**Comment by muhrin** [comment]
Thanks, I'll try and figure out what's going on test side...

File: test/rmq/test_coroutine_communicator.py [hunk 2]
```diff
@@ -197,3 +197,12 @@ def broadcast_subscriber(_comm, _body, _sender=None, _subject=None, _correlation
...
 
 # endregion
+
+
+@pytest.mark.asyncio
+async def test_server_properties(communicator: kiwipy.rmq.RmqCommunicator):
+    props = communicator.server_properties
+    assert isinstance(props, dict)
+
+    assert props['product'] == b'RabbitMQ'
```
Comment (sphuber) [inline test/rmq/test_coroutine_communicator.py:207]: Does it reliably also give the version number across platforms? I guess that is really what we are interested in, in `aiida-core`.

File: kiwipy/rmq/communicator.py (hunk 1 above)
Comment (muhrin) [reply kiwipy/rmq/communicator.py:349]: Done.

File: test/rmq/test_coroutine_communicator.py (hunk 2 above)
Comment (muhrin) [reply test/rmq/test_coroutine_communicator.py:207]: I'm not sure what you mean by 'across platforms' but the RMQ protocol states that the server SHOULD give a version number:

https://www.rabbitmq.com/amqp-0-9-1-reference.html#connection.start.server-properties

I've added this to the docstring...I think we can rely on it.

Comment (muhrin) [reply test/rmq/test_coroutine_communicator.py:207]: Yeah, that's what I thought (re: platforms).  Indeed, to the extent that RMQ conforms to its own spec it should be platform agnostic in this regard...

**Review by sphuber — APPROVED** [review]
Cheers @muhrin

**Pull request merged**